src/main/java/net/dries007/tfc/objects/entity/animal/EntityRabbitTFC.java

```diff
@@ -0,0 +1,456 @@
+	}
+
+	private void updateMoveTypeDuration() {
+		if (this.moveHelper.getSpeed() < 2.2D) {
+			this.currentMoveTypeDuration = 10;
+		} else {
+			this.currentMoveTypeDuration = 1;
+		}
+	}
+
+	private void checkLandingDelay() {
+		this.updateMoveTypeDuration();
+		this.disableJumpControl();
+	}
+
+	protected void jump() {
+		super.jump();
+		double d0 = this.moveHelper.getSpeed();
+
+		if (d0 > 0.0D) {
+			double d1 = this.motionX * this.motionX + this.motionZ * this.motionZ;
+
+			if (d1 < 0.010000000000000002D) {
+				this.moveRelative(0.0F, 0.0F, 1.0F, 0.1F);
+			}
+		}
+
+		if (!this.world.isRemote) {
+			this.world.setEntityState(this, (byte) 1);
+		}
+	}
+
+	@SideOnly(Side.CLIENT)
+	public float getJumpCompletion(float p_175521_1_) {
+		return this.jumpDuration == 0 ? 0.0F : ((float) this.jumpTicks + p_175521_1_) / (float) this.jumpDuration;
+	}
+
+	public void setMovementSpeed(double newSpeed) {
+		this.getNavigator().setSpeed(newSpeed);
+		this.moveHelper.setMoveTo(this.moveHelper.getX(), this.moveHelper.getY(), this.moveHelper.getZ(), newSpeed);
+	}
+
+	public void setJumping(boolean jumping) {
+		super.setJumping(jumping);
+
+		if (jumping) {
+			this.playSound(this.getJumpSound(), this.getSoundVolume(), ((this.rand.nextFloat() - this.rand.nextFloat()) * 0.2F + 1.0F) * 0.8F);
+		}
+	}
+
+	public void startJumping() {
+		this.setJumping(true);
+		this.jumpDuration = 10;
+		this.jumpTicks = 0;
+	}
+
+	public class RabbitJumpHelper extends EntityJumpHelper {
+		private final EntityRabbitTFC rabbit;
+		private boolean canJump;
+
+		public RabbitJumpHelper(EntityRabbitTFC rabbit) {
+			super(rabbit);
+			this.rabbit = rabbit;
+		}
+
+		public boolean getIsJumping() {
+			return this.isJumping;
+		}
+
+		public boolean canJump() {
+			return this.canJump;
+		}
+
+		public void setCanJump(boolean canJumpIn) {
+			this.canJump = canJumpIn;
+		}
+
+		public void doJump() {
+			if (this.isJumping) {
+				this.rabbit.startJumping();
+				this.isJumping = false;
+			}
+		}
+	}
+
+	public void onLivingUpdate() {
+		super.onLivingUpdate();
+
+		if (this.jumpTicks != this.jumpDuration) {
+			++this.jumpTicks;
+		} else if (this.jumpDuration != 0) {
+			this.jumpTicks = 0;
+			this.jumpDuration = 0;
+			this.setJumping(false);
+		}
+	}
+
+	public static void registerFixesRabbit(DataFixer fixer) {
+		EntityLiving.registerFixesMob(fixer, EntityRabbitTFC.class);
+	}
+
+	public void writeEntityToNBT(NBTTagCompound compound) {
+		super.writeEntityToNBT(compound);
+		compound.setInteger("RabbitType", this.getRabbitType());
+	}
+
+	public void readEntityFromNBT(NBTTagCompound compound) {
+		super.readEntityFromNBT(compound);
+		this.setRabbitType(compound.getInteger("RabbitType"));
+	}
+
+	public SoundCategory getSoundCategory() {
+		return SoundCategory.NEUTRAL;
+	}
+
+	@Override
+	public boolean isBreedingItem(ItemStack stack) {
+		return stack.getItem() == Items.CARROT; // TFC carrot?
```

Comment:
Instead, use an ore dict check for "carrot" - add the ore dict values if nessecary

Reply:
Do you mean like OreDictionaryHelper.doesStackMatchOre(stack, "carrot")? Why? Carrot is exact one item, no other 'carrot' items.

Reply:
There are two - vanilla and TFC variants. Potentially more if other mods want to add them. But the simplest reason is this way it becomes modifiable for pack makers instead of hardcoding it.

Reply:
Ok